src/Engine/Trigger/TriggerRepeat.hpp

```diff
@@ -0,0 +1,165 @@
+        /**
+        * @return Pointer to the vector containing name of target functions.
+        */
+        ENGINE_API std::vector<std::string> *GetTargetFunction();
+
+        /// Get if the trigger starts active or not.
+        /**
+        * @return The name of the trigger.
+        */
+        ENGINE_API bool GetStartActive();
+
+        /// Get if the trigger starts active or not.
+        /**
+        * @param value If the trigger starts active or not.
+        */
+        ENGINE_API void SetStartActive(bool value);
+
+        /// Get the delay before the trigger gets active.
+        /**
+        * @return The delay before the trigger gets active.
+        */
+        ENGINE_API float GetDelay();
+
+        /// Set the delay before the trigger gets active.
+        /**
+        * @param value The delay before the trigger gets active.
+        */
+        ENGINE_API void SetDelay(float value);
+
+        /// Get the cooldown before the trigger can get activated again.
+        /**
+        * @return The cooldown before the trigger can get activated again.
+        */
+        ENGINE_API float GetCooldown();
+
+        /// Set the cooldown before the trigger can get activated again.
+        /**
+        * @param value The cooldown before the trigger can get activated again.
+        */
+        ENGINE_API void SetCooldown(float value);
+
+        /// Get the amount of times a trigger can be activated before it is no longer active.
+        /**
+        * @return The amount of times a trigger can be activated before it is no longer active.
+        */
+        ENGINE_API float GetTriggerCharges();
+
+        /// Set the amount of times a trigger can be activated before it is no longer active.
+        /**
+        * @param value The amount of times a trigger can be activated before it is no longer active.
+        */
+        ENGINE_API void SetTriggerCharges(int value);
+
+        /// Vector containing target entities.
+        /**
+        * @return Pointer to the vector containing target entities.
+        */
+        ENGINE_API std::vector<Entity*> *GetTargetEntity();
+
+        /// Vector containing collided entities.
+        /**
+        * @return Pointer to the vector containing collided entities.
+        */
+        ENGINE_API std::vector<Entity*> *GetCollidedEntity();
+
+        /// Vector containing the event data.
+        /**
+        * @return Pointer to the vector containing event data.
+        */
+        ENGINE_API std::vector<EventStruct>* GetEventVector();
+
+        /// Get the entity this component belongs to.
+        /**
+        * @return The entity this component belongs to.
+        */
+        ENGINE_API Entity* GetOwningEntity();
+
+        /// Set the entity this component belongs to.
+        /**
+        * @param value The entity this component belongs to.
+        */
+        ENGINE_API void SetOwningEntity(Entity* value);
+                
+        ENGINE_API void Process() override;
```

Comment:
Undocumented methods.

Reply:
Is it okay to just refer to the SuperClass which they override?

Reply:
I think so.